etna/analysis/forecast/plots.py

```diff
@@ -731,7 +749,10 @@ def plot_metric_per_segment(
             metrics_df.groupby("segment").agg({metric_name: aggregation_mode.get_function()}).to_dict()[metric_name]
         )
     else:
-        metrics_dict = metrics_df["segment", metric_name].to_dict()[metric_name]
+        metrics_dict = metrics_df[["segment", metric_name]].to_dict()[metric_name]
```

Comment:
Maybe modify a bit to exclude manual filtering below?

Aggregation above would require similar changes.

```suggestion
        metrics_dict = metrics_df[["segment", metric_name]].set_index("segment").dropna().to_dict()[metric_name]
```

Reply:
It also fixes current error that in this else clause we don't have segments as keys.